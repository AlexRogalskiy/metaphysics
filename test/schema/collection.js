const gravityData = {
  id: "saved-artwork",
  name: "Saved Artwork",
  default: true,
  description: "",
  image_url: null,
  image_versions: null,
  private: false
};

describe("Collections", () => {
  describe("Handles getting collection metadata", () => {
    const Collection = schema.__get__("Collection");

    beforeEach(() => {
      const gravity = sinon.stub();
      gravity.with = sinon.stub().returns(gravity);
      gravity
        .withArgs("collection/saved-artwork", { user_id: null })
        .returns(Promise.resolve(gravityData));

      Collection.__Rewire__("gravity", gravity);
    });

    afterEach(() => {
      Collection.__ResetDependency__("gravity");
    });

    it("returns collection metadata", () => {
      const query = `
        {
          collection(id: "saved-artwork") {
            name
            private
            default
          }
        }
      `;
      return runQuery(query).then(data => {
        expect(data).toMatchSnapshot();
      });
    });
  });
});
