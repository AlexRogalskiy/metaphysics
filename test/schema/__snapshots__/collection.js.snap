// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collections Handles getting collection metadata returns collection metadata 1`] = `
Object {
  "collection": Object {
    "default": true,
    "name": "Saved Artwork",
    "private": false,
  },
}
`;
